### DISCUSSION TITLE
How to update test status in "-results.json" programatically when a test is marked as ignored based on env variable

### DISCUSSION THREAD

**Discussion by singh-swati28** [allure-framework/allure2#2147] [Questions & Support] [answered]
Hi @baev 

I am using Allure Specflow with xUnit.
Problem Statement: I am marking my test scenarios as 'ignored' dynamically by using IUnitTestRuntimeProvider based on few environment variables passed at runtime.  
`_unitTestRuntimeProvider.TestIgnore("This` test was intentionally `skipped");`
On executing above line, Specflow is marking Step as Failed due to Xunit.SkipException and hence Allure results.json files are also marking the test status as Failed.

These tests are ignored intentionally, so I am curious, if there is a way to set Status as 'skipped' in Allure "-results.json" file for these specific test scenarios/steps

Appreciate your help!.

**Comment by neparij** [comment]
@delatrie take a look, please

**Reply by singh-swati28** [reply, marked as answer]
@neparij, I was able to update testcase status in "results.json" file by calling AllureLifeCycle.Instance.UpdateTestCase function.
Thank you for your help!
